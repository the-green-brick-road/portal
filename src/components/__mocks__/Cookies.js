/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
