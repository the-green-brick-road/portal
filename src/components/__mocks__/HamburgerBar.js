/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
