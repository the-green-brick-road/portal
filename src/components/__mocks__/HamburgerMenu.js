/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;