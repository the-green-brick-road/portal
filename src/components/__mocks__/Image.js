/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
