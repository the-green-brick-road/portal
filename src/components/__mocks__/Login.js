/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
