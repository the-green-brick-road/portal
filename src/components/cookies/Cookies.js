/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Cookie component
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @23 june 2023
# Latest revision: 23 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }                            from 'react';

/* Material UI includes */
import { Button, Typography, Link, Paper }     from '@mui/material';

/* Website includes */
import { useLogging, useAnalytics, useDesign } from '../../providers';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color }                 = props;
    const { deactivatePerformance, activatePerformance, deactivateAnalytics, activateAnalytics  } = useAnalytics();
    const { logText, onRender }     = useLogging();
    const { setHasAcceptedCookies } = useDesign();
    const componentName             = 'Cookies';

    /* ------ Manage cookie change ----- */
    const activateCookies = () => {

        logText(componentName, 'debug', 'workflow', ' activateCookies');
        setHasAcceptedCookies(true);
        activateAnalytics();
        activatePerformance();

    };
    const deactivateCookies = () => {

        logText(componentName, 'debug', 'workflow', ' deactivateCookies');
        setHasAcceptedCookies(false);
        deactivateAnalytics();
        deactivatePerformance();

    };

    /* ----------- Define HTML --------- */
    return (

        <Profiler id={componentName} onRender={onRender}>
            <Paper id="cookies" square={true} style={{ backgroundColor:color, position:'relative', paddingLeft:10, paddingRight:10, paddingBottom:10, paddingTop:10, bottom:0 }}>
                <Typography style= {{ color:'#ffffff' }}>
                    We use cookies to enable the proper functioning and security of our websites, and help us offer you the best possible user experience. By clicking Accept, you consent to the use of these cookies for advertising and analytics. You can change your cookie settings at any time. For more information, please read our
                    <Link href="/policy" style= {{ color:'#ffffff' }}>
                        cookie policy
                    </Link>
                </Typography>
                <Button aria-label='accept' onClick={activateCookies} style={{ backgroundColor: '#ffffff', margin:'10px' }} >
                    <Typography style={{ color: color, fontWeight:'bold' }}>
                        Accept
                    </Typography>
                </Button>
                <Button aria-label='decline' onClick={deactivateCookies} style={{ backgroundColor: '#ffffff', margin:'10px' }} >
                    <Typography style={{ color: color, fontWeight:'bold' }}>
                        Decline
                    </Typography>
                </Button>
            </Paper>
        </Profiler>
    );

}

export default Cookies;
