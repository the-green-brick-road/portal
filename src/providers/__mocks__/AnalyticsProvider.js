/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2024] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Analytics provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @13 february 2024
# Latest revision: 13 february 2024
# ---------------------------------------------------- */

/* React includes */
import { createContext } from 'react';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isAnalyticsActivated :   false,
    isPerformanceActivated : false,
    activatePerformance:     jest.fn(() => { state.isPerformanceActivated = true }),
    deactivatePerformance:   jest.fn(() => { state.isPerformanceActivated = false }),
    activateAnalytics:       jest.fn(() => { state.isAnalyticsActivated = true }),
    deactivateAnalytics:     jest.fn(() => { state.isAnalyticsActivated = false }),
};


function AnalyticsProvider(props) {

    const { children, isAnalyticsActivated = false, isPerformanceActivated = false } = props;

    state.isAnalyticsActivated      = isAnalyticsActivated
    state.isPerformanceActivated    = isPerformanceActivated

    /* ----------- Define HTML --------- */
    return (
        <context.Provider value={state}>
            {children}
        </context.Provider>
    );

}

function useAnalytics() { return state; }

export { useAnalytics, AnalyticsProvider };