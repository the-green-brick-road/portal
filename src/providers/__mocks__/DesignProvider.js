/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Design provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

/* Material UI includes */
import { CssBaseline }                                 from '@mui/material';
import { ThemeProvider, createTheme }                  from '@mui/material/styles';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isWebpSupported: true,
    isSliding:       false,
    isDarkMode:      false,
    images:          {},
    screen:          '',
    sizes:           {},
    setIsSliding:    jest.fn((value) => { state.isSliding = value }),
};

function DesignProvider(props) {

    const { children, isWebpSupported, screen, isDarkMode = false, isSliding = false, sizes, images = [], theme = {} } = props;

    /* ------ Defining behaviour ------- */
    var local_theme = {}
    if (Object.keys(theme).length !== 0) { local_theme = createTheme(theme); }

    state.isWebpSupported = isWebpSupported;
    state.screen          = screen;
    state.images          = {}
    state.sizes           = sizes;
    state.isSliding       = isSliding;
    state.isDarkMode      = isDarkMode;
    for (const image of images)
    {

        const rawext = image["raw"].split('.').pop();
        const webext = image["web"].split('.').pop();

        state.images[image.name] = {}
        state.images[image.name][rawext] = {}
        state.images[image.name][webext] = {}

        state.images[image.name][rawext]['small']           = {}
        state.images[image.name][rawext]['small']['img']    = require(`../../assets/small/${image.raw}`)
        state.images[image.name][rawext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][rawext]['medium']          = {}
        state.images[image.name][rawext]['medium']['img']   = require(`../../assets/medium/${image.raw}`)
        state.images[image.name][rawext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][rawext]['large']           = {}
        state.images[image.name][rawext]['large']['img']    = require(`../../assets/${image.raw}`)
        state.images[image.name][rawext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name][webext]['small']           = {}
        state.images[image.name][webext]['small']['img']    = require(`../../assets/small/${image.web}`)
        state.images[image.name][webext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][webext]['medium']          = {}
        state.images[image.name][webext]['medium']['img']   = require(`../../assets/medium/${image.web}`)
        state.images[image.name][webext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][webext]['large']           = {}
        state.images[image.name][webext]['large']['img']    = require(`../../assets/${image.web}`)
        state.images[image.name][webext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name]['default']= require(`../../assets/${image.raw}`)

    }

    /* ----------- Define HTML --------- */
    return (
        <ThemeProvider theme={local_theme}>
            <CssBaseline />
            <context.Provider value={state}>
                {children}
            </context.Provider>
        </ThemeProvider>
    );

}

function useDesign() { return state; };

export { useDesign, DesignProvider };