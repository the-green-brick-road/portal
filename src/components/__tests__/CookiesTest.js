/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Footer component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM, act, screen, fireEvent } from '@testing-library/react'
import { expect, test}                               from '@jest/globals';

/* Component under test */
import { default as Cookies }  from '../cookies/Cookies';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { useDesign as mockUseDesign, DesignProvider as MockDesignProvider } from '../../providers/__mocks__/DesignProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../../providers/__mocks__/LoggingProvider';
import { useAnalytics as mockUseAnalytics, AnalyticsProvider as MockAnalyticsProvider } from '../../providers/__mocks__/AnalyticsProvider';
jest.mock('../../providers', () => ({
    useLogging: (() => { return mockUseLogging(); }),
    useDesign: (() => { return mockUseDesign(); }),
    useAnalytics: (() => { return mockUseAnalytics(); }),
}));
/* eslint-enable jest/no-mocks-import */

describe("Cookies component" ,() => {

    const theme = {
        "palette": {
            "common": {
                "black": "#444444",
                "white": "#555555",
            },
            "primary": {
                "main": "#666666",
                "light": "#666667",
                "dark": "#666668",
            },
            "secondary" : {
                "main": "#777777",
                "light": "#777778",
                "dark": "#777779",
            },
            "disabled" : {
                "main": "#888888",
                "light": "#888889",
                "dark": "#888880",
            },
            "text" : {
                "primary": "#111111",
                "secondary": "#222222",
                "disabled": "#333333",
            },
        },
        "direction": "ltr",
        "typography": {
            "h1": {
                "fontSize": "2.5rem",
                "color": "#111111",
                "textTransform": "uppercase",
                "fontFamily": "Roboto Condensed",
            },
            "h2": {
                "fontSize": "2rem",
                "color": "#222222",
                "textTransform": "lowercase",
                "fontFamily": "Roboto Condensed",
            },
            "body1": {
                "fontSize": "14px",
                "color": "black",
                "fontFamily": "Roboto",
                "position": "relative",
                "textAlign": "justify",
            },
        },
    }

    test('Should enable cookies activation and deactivation', async () => {

        const state_analytics = mockUseAnalytics();
        const state_design = mockUseDesign();

        let view = null
        await act(async () => { // eslint-disable-line testing-library/no-unnecessary-act

            view = render(
                <div>
                    <MockAnalyticsProvider>
                        <MockLoggingProvider>
                            <MockDesignProvider isWebpSupported={true} screen='large' theme={theme} hasAcceptedCookies={false}>
                                <Cookies color='white'/>
                            </MockDesignProvider>
                        </MockLoggingProvider>
                    </MockAnalyticsProvider>
                </div>
            );

        })

        const delay = ms => new Promise(res => setTimeout(res, ms));
        await delay(100);

        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();
        expect(state_design.setHasAcceptedCookies).toHaveBeenCalledTimes(0)
        expect(state_analytics.activatePerformance).toHaveBeenCalledTimes(0)
        expect(state_analytics.activateAnalytics).toHaveBeenCalledTimes(0)
        expect(state_analytics.deactivatePerformance).toHaveBeenCalledTimes(0)
        expect(state_analytics.deactivateAnalytics).toHaveBeenCalledTimes(0)

        await act(async () => {fireEvent.click(screen.getByRole('button', { name: 'accept' }))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state_design.setHasAcceptedCookies).toHaveBeenLastCalledWith(true)
        expect(state_design.setHasAcceptedCookies).toHaveBeenCalledTimes(1)
        expect(state_analytics.activatePerformance).toHaveBeenCalledTimes(1)
        expect(state_analytics.activateAnalytics).toHaveBeenCalledTimes(1)
        expect(state_analytics.deactivatePerformance).toHaveBeenCalledTimes(0)
        expect(state_analytics.deactivateAnalytics).toHaveBeenCalledTimes(0)

        await act(async () => {fireEvent.click(screen.getByRole('button', { name: 'decline' }))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state_design.setHasAcceptedCookies).toHaveBeenLastCalledWith(false)
        expect(state_design.setHasAcceptedCookies).toHaveBeenCalledTimes(2)
        expect(state_analytics.activatePerformance).toHaveBeenCalledTimes(1)
        expect(state_analytics.activateAnalytics).toHaveBeenCalledTimes(1)
        expect(state_analytics.deactivatePerformance).toHaveBeenCalledTimes(1)
        expect(state_analytics.deactivateAnalytics).toHaveBeenCalledTimes(1)

    })

})
