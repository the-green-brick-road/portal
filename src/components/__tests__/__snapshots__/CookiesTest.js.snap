// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookies component Should enable cookies activation and deactivation 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 css-1p2ttux-MuiPaper-root\\"
        id=\\"cookies\\"
        style=\\"background-color: white; position: relative; padding: 10px 10px 10px 10px; bottom: 0px;\\"
      >
        <p
          class=\\"MuiTypography-root MuiTypography-body1 css-15v4sip-MuiTypography-root\\"
          style=\\"color: rgb(255, 255, 255);\\"
        >
          We use cookies to enable the proper functioning and security of our websites, and help us offer you the best possible user experience. By clicking Accept, you consent to the use of these cookies for advertising and analytics. You can change your cookie settings at any time. For more information, please read our
          <a
            class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-uzv0uc-MuiTypography-root-MuiLink-root\\"
            href=\\"/policy\\"
            style=\\"color: rgb(255, 255, 255);\\"
          >
            cookie policy
          </a>
        </p>
        <button
          aria-label=\\"accept\\"
          class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-jnjbjn-MuiButtonBase-root-MuiButton-root\\"
          style=\\"background-color: rgb(255, 255, 255); margin: 10px;\\"
          tabindex=\\"0\\"
          type=\\"button\\"
        >
          <p
            class=\\"MuiTypography-root MuiTypography-body1 css-15v4sip-MuiTypography-root\\"
            style=\\"color: white; font-weight: bold;\\"
          >
            Accept
          </p>
          <span
            class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
          />
        </button>
        <button
          aria-label=\\"decline\\"
          class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-jnjbjn-MuiButtonBase-root-MuiButton-root\\"
          style=\\"background-color: rgb(255, 255, 255); margin: 10px;\\"
          tabindex=\\"0\\"
          type=\\"button\\"
        >
          <p
            class=\\"MuiTypography-root MuiTypography-body1 css-15v4sip-MuiTypography-root\\"
            style=\\"color: white; font-weight: bold;\\"
          >
            Decline
          </p>
          <span
            class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
          />
        </button>
      </div>
    </div>
  </div>
</body>"
`;
